app.py

```diff
@@ -59,6 +59,26 @@ def node_register():
     return jsonify({})
 
 
+@app.route('/v1/proposals', methods=['GET'])
+def proposals():
+    node_key = request.args.get('node_key')
+
+    if node_key:
+        node = Node.query.get(node_key)
+        if not node:
+            return jsonify(error='node by node_key not found'), 400
```

Comment:
maybe `404 not found` is more specific in this case

Reply:
According to rest if you have list of resources (proposals) and applying filter provides empty list its perfectly ok to return 200 status and empty list without any errors.

Reply:
Updated endpoint to return empty list in this case.